HIP/hip-583.md

```diff
@@ -10,220 +10,452 @@ status: Accepted
+
+An `Account` also has a system-assigned account identifier of the form `shard.realm.num`, where each value is an 8-byte
+number. An `Account` may also be referred to with an identifier of the form `shard.realm.alias`, where the alias is of
+one of the following forms:
+
+1. `account-num alias` - This is the Hedera account number styled as an Ethereum account hex string. The
+   Hedera account number is hex encoded and the value is prefixed with zeros to form a 20 byte address. This format
+   cannot be used to auto-create an account, but can be used wherever an `alias` can be used. This is sometimes
+   referred to as the "long-zero" alias, since the leftmost 12 bytes are 0.
+2. `key alias` - This format is created by using the bytes of a simple cryptographic public key supported by Hedera
+   (such as `ED25519` or `ECDSA secp256k1`). It is not necessary for the account's key to match the `alias`, only that
+   the account owner has asserted cryptographic proof of ownership of the account `alias` key in addition to the
+   account key. The key is protobuf-encoded and those raw bytes are included in the `alias`.
+3. `evm address alias` - This format is created by using the rightmost 20 bytes of the 32 byte `Keccak-256` hash of an
+   `ECDSA secp256k1` public key. This calculation is in the manner described by the Ethereum Yellow Paper.
+   Note, that the recovery id is not formally part of the public key and not included in the hash. This is
+   calculated on the consensus nodes using the `ECDSA` key provided in the auto create `CryptoTransfer` transaction.
+   While both `evm address alias` and `account-num alias` are 20 byte values, they can be differentiated by the fact
+   that `account-num alias` is always prefixed with 12 bytes. The likelihood of randomly generated ECDSA keys, being
+   hashed, having 12 sequential zeros is very low. Those keys, should they be found, cannot be used on the network.
+
+From the perspective of the Ethereum ecosystem, the `account-num alias` and `evm address alias` are both
+directly usable when prefixed with `0x`. Examples of each format are noted below:
+(Please NOTE: the `alias` is a binary format when encoded on-wire for protobuf. We represent the bytes as hexidecimal
+ASCII for clarity in the documentation, but in reality these are raw bytes)
+
+| Identifier Type           | Account ID Example (Protobuf)                                            | Ethereum Address Example (String)          |
+|---------------------------|--------------------------------------------------------------------------|--------------------------------------------|
+| Hedera Account ID         | 0.0.10                                                                   | N/A                                        |
+| Hedera Account Num Alias  | 0.0.000000000000000000000000000000000000000a                             | 0x000000000000000000000000000000000000000a |
```

Comment:
We need to document clearly that in the extremely rare chance that you have a private key that produces an evm address with 12 leading zero bytes (96 bits!!)... I'm sorry. It won't work on our network properly.